program/src/instructions/shared_accounts.rs

```diff
@@ -520,16 +526,33 @@ pub struct MplxShared<'info> {
     pub mint: Box<InterfaceAccount<'info, Mint>>,
 
     /// The Token Metadata metadata account of the NFT.
-    /// CHECK: ownership, structure and mint are checked in assert_decode_metadata.
-    #[account(mut)]
+    /// CHECK: ownership, structure and mint are checked in assert_decode_metadata, seeds checked here.
+    #[account(mut,
+    seeds = [
+        Metadata::PREFIX,
+        mpl_token_metadata::ID.as_ref(),
+        mint.key().as_ref(),
+    ],
+    bump,
+    seeds::program = mpl_token_metadata::ID,
+    )]
```

Comment:
Is seeds checking added back in here for redundancy? Don't we CPI into tmeta anyways which will do the check?

Reply:
Yes, recommended by audit report findings even though Metadata should be checked in Token Metadata in all cases.